libs/substrate-differ/src/differs/reduced/reduced_differ.rs

```diff
@@ -63,34 +65,48 @@ impl Differ for ReducedDiffer {
 		let r2 = &self.r2;
 
 		// We gather the Set of all indexes in both pallets
-		let indexes_1: Vec<Index> = r1.pallets.iter().map(|pallet| pallet.index).collect();
-		let indexes_2: Vec<Index> = r2.pallets.iter().map(|pallet| pallet.index).collect();
-		let mut indexes: HashSet<Index> = HashSet::new();
-		indexes.extend(indexes_1.iter());
-		indexes.extend(indexes_2.iter());
+		let indexes_1: HashMap<(String, Index), &ReducedPallet> =
```

Comment:
This `(String, Index)` shows in a few places. Let's extract it so we can have the freedom to easily fix the type as needed.